graylog2-server/src/test/java/org/graylog/testing/containermatrix/ContainerMatrixTestEngine.java

```diff
@@ -98,12 +99,57 @@ private Set<Class<? extends PluginJarsProvider>> getPluginJarsProvider(Set<Class
         return get(annotatedClasses, (ContainerMatrixTestsConfiguration annotation) -> Stream.of(annotation.pluginJarsProvider()));
     }
 
-    private Set<SearchServer> getSearchServerVersions(Set<Class<?>> annotatedClasses) {
+    /**
+     * Property has to be in the form of Elasticsearch|OpenSearch-x.x.x
+     *
+     * @return
+     */
+    private static String getSearchVersionOverride() {
+        return System.getProperty("graylog.matrix.tests.search.version.override");
```

Comment:
What about doing all the splitting, parsing and comparisions here and return Optional<SearchVersion> that could be then used elsewhere?